packages/charts/src/views/chart.view.ts

```diff
@@ -43,9 +43,9 @@ export abstract class ChartView<ChartProps extends ChartViewProps<any>,
     this.require(RequiredViewChanges.DRAW);
   }
 
-  abstract getCanvasPositionForPoint(point: XYPoint): CanvasPosition;
+  abstract getCanvasPositionForPoint(point: XYPoint<any>): CanvasPosition;
 
-  abstract getPointForCanvasPosition(position: XYPoint): XYPoint | undefined;
+  abstract getPointForCanvasPosition(position: XYPoint<number>): XYPoint<number> | undefined;
```

Comment:
Since that is a major change that touches all views derived from this one, I'm thinking of adding one more generic param to ChartView, so that you could only modify AreaChart and leave other components unchanged:

```ts
export abstract class ChartView<ChartProps extends ChartViewProps<any>,
  Style = NonNullable<ChartProps['style']>, YType = number> extends View<ChartProps> {
   // ...

  abstract getCanvasPositionForPoint(point: XYPoint<YType>): CanvasPosition;

  abstract getPointForCanvasPosition(position: XYPoint<YType>): XYPoint<YType> | undefined;

   // ...

}
```

WDYT?

Reply:
Similarly, `DataContainer` should become more flexible to accommodate this change - it probably can be achieved by inheritance, yet I'm aware that improving the architecture of Kanva is probably not what you want to work on, so I'll accept whatever @DawidYerginyan suggests on that topic 😉

Reply:
I haven't had a look at this MR yet, I have to go through it, but I'd definitely want to make the XYPoint generic If that's possible. Even tho this MR is not about the architecture of Kanva, It still is very important to do things the right way so that we're not screwed later on 😄 .

Reply:
When I started this task I approached it in a very similar way, I tried to incorporate the generic type in `DataContainer` definition as such:

```typescript
export class DataContainer<DataPoint = any, YPoint = number> {
...
```
which was fine at the beginning, but then I realized the `pointAccessor` and `getNormalizedData` would have to incorporate such change, I don't have a solution yet, but I might try to fix it if you let me. I had a vision that while any chart actually consumes the data container they could somehow infer the proper typing? Would be also really cool to lock certain `DataDisplayType` to internal data shapes.

/// 
also tried with such approach:

```typescript
export class DataContainer<InputDataPoint = any, OutputDataPoint extends XYPoint<any> = XYPoint> {
...
```
but didn't really succeed with properly typing this throughout packages, but might be a good starting point for discussion.

Reply:
Let's move this to a discussion thread 🙂